libyara/object.c

```diff
@@ -609,6 +609,28 @@ int yr_object_copy(
       break;
 
     case OBJECT_TYPE_ARRAY:
+      ;
+      YR_OBJECT_ARRAY *array = (YR_OBJECT_ARRAY *) object;
```

Comment:
Declaring variable inside switch statements doesn't work in C89. You can use the "o" variable.